packages/utils/src/dom/get-element-by-url.ts

```diff
@@ -2,7 +2,13 @@ import { URL } from 'url';
 
 import { HTMLDocument, HTMLElement } from './html';
 
-const getSrcsetUrls = (srcset: string): string[] => {
+/**
+ * Two level cache: doc.base / url
+ * TODO: Use quick-lru so that it doesn't grow without bounds
+ */
+const CACHED_URLS: { [key: string]: { [key: string]: URL } } = { };
```

Comment:
This is great! One request: I'd prefer to use nested `Map`s here unless there's a particular reason you didn't. [Browser support for it](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Browser_compatibility) is plenty good enough for what we're targeting. I think that should give us a slight additional performance edge too (though probably not a significant one).